index.js

```diff
@@ -38,14 +39,28 @@ function getLayerValues(layer, key) {
     return values;
 }
 
-function decorateLayer(layer, keysToKeep, newProps, requiredKeys) {
+function filterByKeys(layer, requiredKeys) {
+    var keys = layer.keys;
+    var requiredLookup = {};
+    var requiredCount = (requiredKeys || []).length;
+
+    if (requiredCount) {
+        for (var i = 0; i < requiredCount; i++) {
+            requiredLookup[keys.indexOf(requiredKeys[i])] = true;
+        }
+
+        layer.features = layer.features.filter(function (feature) {
+            return hasAllKeys(feature.tags, requiredLookup, requiredCount);
+        });
+    }
```

Comment:
I've removed the condition, but`requiredCount` is used by `hasAllKeys()` to determine if all the required keys were found